### DISCUSSION TITLE
Only disable GitHub comments on main branch

### DISCUSSION THREAD

**Discussion by IGassmann** [vercel/vercel#9066] [Ideas]
[`github.silent`](https://vercel.com/docs/project-configuration#git-configuration/github-silent) allows us to stop the Vercel bot from commenting on pull requests and commits. However, it doesn't let you decide only to disable comments on the `main` branch.

Since GitHub already has [commit status checks](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/collaborating-on-repositories-with-code-quality-features/about-status-checks) with a link to the deployment, comments on the `main` are superfluous. They just spam the repository contributors' GitHub inbox every time there's a commit.

That being said, comments on PR are still useful because they provide an easy access link to the deployment to be used for review and also information about Deployment Comments.

That's why it would be great to allow us only to disable those comments on the commits of the `main` branch.

**Comment by amyegan** [comment]
Are you suggesting something like [`git.deploymentEnabled`](https://vercel.com/docs/project-configuration#git-configuration/git-deployment-enabled), but for comments instead of deployments?

**Reply by IGassmann** [reply]
That's right :)

**Reply by amyegan** [reply]
Thanks for sharing this idea. I sent it to the team 🙂

**Reply by tomislavhrenmp** [reply]
Any progress on this one? It would be great if we could disable it.

**Comment by jonworek** [comment]
I came across this discussion when searching for how to disable the vercel bot comments on the PR. It looks like this functionality is now built into the project settings UI, and is deprecated from `vercel.json`.

See: https://vercel.com/docs/deployments/git/vercel-for-github#silence-github-comments

I have two options visible to me: 

![image](https://github.com/vercel/vercel/assets/3124548/9fab8cda-1d55-452c-9bb2-a7c81571fa15)

"Pull Request Comments" was the setting I was looking to disable.

"Commit Comments" looks like it might be what you're looking for @IGassmann .

**Reply by IGassmann** [reply]
Unfortunately, as @samthomson mentioned, this configuration doesn't seem to work.

**Comment by samthomson** [comment]
<img width="951" alt="image" src="https://github.com/vercel/vercel/assets/6588340/9ac13c60-417d-4134-935e-a2412ad64a8d">

Why do I still get a commit comment for every push to master?

**Comment by samthomson** [comment]
@amyegan can you confirm if this is the intended behaviour - that I can't disable vercel bot comments on commits this way, as it seems from the toggle/options that it ought to be disable-able. Or how can I resolve this (aside from disabling all github notifications - which is not something I'm prepared to do).

**Reply by amyegan** [reply]
@samthomson @IGassmann That is not expected behavior. Thanks for reporting it. 

In the interest of gathering info to aid investigation, I have a couple of questions:
- Does the project happen to be part of a monorepo?
- What is the name or a deployment link of the project?

**Reply by samthomson** [reply]
Sure.
- The project is not part of a mono repo. There is one github repo, _but_ there are two vercel projects that get build from it (each governed by different env vars).
- the repo has two projects attached: 1. guidedtrekking 2. guided-peaks 
- Interestingly, building the first project doesn't result in the vercel bot commenting to the repo - only with the second project. From what I can see both vercel projects have the same settings.
